api/common/utilities.py

```diff
@@ -20,13 +21,12 @@ def from_string(cls, value: str):
         return cls(value)
 
 
-def handle_version_retrieval(domain, dataset, version) -> int:
-    if not version:
-        AppLogger.info(
-            "No version provided by the user. Retrieving the latest version from the crawler."
-        )
-        version = aws_resource_adapter.get_version_from_crawler_tags(domain, dataset)
-    return version
+def construct_dataset_metadata(
+    layer: Layer, domain: str, dataset: str, version: Optional[int] = None
+) -> DatasetMetadata:
+    dataset = DatasetMetadata(layer, domain, dataset, version)
+    dataset.handle_version_retrieval(aws_resource_adapter)
```

Comment:
I don't think this makes sense. Are we meant to be returning the `dataset.handle_version_retrieval` function?